File: src/model/Stats.java
package model;

public class Stats {
    private int gols;
    private int assits;
    private int partidas;
    private int quantMvp;
    private int soma;
    private int media;


    public int getGols() {
        return gols;
    }

    public void setGols(int gols) {
        this.gols = gols;
    }

    public int getAssits() {
        return assits;
    }

    public void setAssits(int assits) {
        this.assits = assits;
    }

    public int getPartidas() {
        return partidas;
    }

    public void setPartidas(int partidas) {
        this.partidas = partidas;
    }

    public int getQuantMvp() {
        return quantMvp;
    }

    public void setQuantMvp(int quantMvp) {
        this.quantMvp = quantMvp;
    }

    public void mediaGA(){
        media = soma/partidas;
    }

    public void somaGA(){
        soma = gols + assits;
    }

    @Override
    public String toString() {
        return new StringBuilder().append("Stats{").append("Gols: ").append(gols).append(", Assistencias: ").append(assits).append(", Partidas: ").append(partidas).append(", MVPs: ").append(quantMvp).append(", Soma G/A: ").append(soma).append("Media G/A: ").append(media).append('}').toString();
    }
}

File: sgfc_too/lib/Stats.dart
class Stats {
  int _gols = 0;
  int _assists = 0;
  int _partidas = 0;
  int _quantMvp = 0;
  int _soma = 0;
  int _media = 0;

  // Getters e setters
  int get gols => _gols;
  set gols(int value) => _gols = value;

  int get assists => _assists;
  set assists(int value) => _assists = value;

  int get partidas => _partidas;
  set partidas(int value) => _partidas = value;

  int get quantMvp => _quantMvp;
  set quantMvp(int value) => _quantMvp = value;

  int get soma => _soma;
  int get media => _media;

  // Métodos
  void calcularSomaGA() {
    _soma = _gols + _assists;
  }

  void calcularMediaGA() {
    if (_partidas > 0) {
      _media = _soma ~/ _partidas; // ~/ = divisão inteira
    } else {
      _media = 0;
    }
  }

  @override
  String toString() {
    return 'Stats{Gols: $_gols, Assistências: $_assists, Partidas: $_partidas, MVPs: $_quantMvp, Soma G/A: $_soma, Média G/A: $_media}';
  }
}
